### DISCUSSION TITLE
A survey for tabularray users

### DISCUSSION THREAD

**Discussion by lvjr** [lvjr/tabularray#119] [General]
Welcome to `tabularray` Discussions! Here is a survey for you:

1. How did you know of `tabularray` package?
2. Which TeX distribution (TeX Live, MiKTeX, etc) and which version do you use?
3. Which parts of the [manual](http://mirrors.ctan.org/macros/latex/contrib/tabularray/tabularray.pdf) of `tabularray` package do you think need to be improved?

Please fill out the survey in your comment. Thank you!

**Comment by leomayer** [comment]
1. From your reply on `tex.stackexchange`
2. `texlive` with OpenSusue ==> your package is NOT yet provided by the packaging structure - what a pity.
3. Already responded in another ticket.

**Comment by cdacl** [comment]
1. From your reply on tex.stackexchange.
2. `texlive` in Win10, and `mactex` in macos.
3. The same as leomayer.

**Comment by syvshc** [comment]
1. from QQ group user Nan Geng
2. TeXLive2021 on win10 and ArchLinux
3. prefect I think

**Comment by bluebear94** [comment]
1. From tex.stackexchange
2. TeX Live 2021 (on Manjaro Linux)
3. Point out some caveats when migrating from another table package; maybe a troubleshooting section

**Comment by mafsi** [comment]
1. From your reply on tex.stackexchange.com
2. MikTeX on Win 10
3. The manual is written for those who already know how to use table. This is not my case. I jumped directly to `tabularray` being a novice in LaTeX. `Tabularray` is a great package and deserve an extended manual for novices. 
    -  Please improve section `1.6 Column Types`.
    -  Also, [this part is not covered](https://tex.stackexchange.com/questions/614040/is-there-an-easy-way-to-create-tables-for-educational-materials#comment1554089_614130) in manual.
  Thank you for the package!

**Comment by mdalimran-abir** [comment]
1. Can't exactly remember. Most probably through an answer in the TeX SE.
2. MikTeX 22.3
3. Much friendlier for novices compared to many other documentations.

**Comment by JeT76-ESCP** [comment]
1. Zarko on TeX.SE
2. MikTeX
3. You deserve the same [webpage](http://pgfplots.sourceforge.net/gallery.html) for your examples :)

**Comment by registor** [comment]
1. from easybook class source codes
2. TeXLive2022 on Ubuntu20.04
3. prefect I think

**Comment by mauriziopassariello** [comment]
1. From stackexchange 
2. TexLive 2022
3. The documentation assumes you know how to put together the table. It is not friendly for novices. A quick guide with examples or a cheat sheet may be useful

**Comment by ysalmon** [comment]
1. By chance while browsing things about LaTeX3.
2. TexLive2022 under Linux.
3. What @mafsi and @mauriziopassariello said. To let tabularray replace the other packages and the base ``array`` (which imo it should because it is much better), a documentation is needed that does not rely on knowledge of those. Also, limitations of the package (currently, inability to have ``&`` or ``%`` in the table, inter alia) should be laid out in the doc.

**Comment by xfz329** [comment]
1recommended from a latext project.
2 latex 2023.
3 how to use the theme and the template. #394

**Comment by abbas-shams** [comment]
1. When searching for a solution to my problem of merging multiple rows and columns simultaneously.
2. TeXlive 2023 under both Windows and Linux
3. Unfortunately, there is not any step-by-step guideline for `tabularray` anywhere; so, in addition to the package documentation (which is not, in my opinion, a manual), the developers should provide a step-by-step guideline for those starting to learn LaTeX. My question is, why a user should learn multiple packages with the same aim? The documentation for `tabularray` has been provided based on having a basic knowledge of typesetting tables with macros other than what `tabularray` has provided. It's a waste of time. Users should be able to jump into `tabularray` without knowing basic information about typesetting tables by other means.

Anyhow, I should thank from developers of `tabularray` for their efforts and for developing such a powerful package.

Also, I have a suggestion. There is `longtblr` environment for tables whose height exceeds the height of the page. Some tables are wide (and maybe simultaneously tall). For such tables, another environment (called `widetable`) may be produced to spread columns of tables on multiple pages. An example of such wide tables can be found in the standard "[ASME BPVC Section II-Materials-Part D-Properties](https://www.asme.org/codes-standards/find-codes-standards/bpvc-iid-bpvc-section-ii-materials-part-d-properties-(1)/2023/print-book)". In this document, table columns have been spread over four pages.

**Comment by celdorwow** [comment]
Welcome to tabularray Discussions! Here is a survey for you:

1. I came across `tabularray` when reading answers on `TeX.SE`.
2. Current, at the time of writing, TeX Live 2023.
3. IMO the part related to `longtblr` and `talltblr` are particularly confusing. It is not simple and straightforward to figure out what one should do to customise these environments. I understand the idea is to define custom styles components and then utilise such "elements"  in different styles. A number of styles can  utilise the same style component and the one style can have a number of components. This adds flexibility to define multiple styles without necessarily repeating the code etc. This is great idea. It is also understandable it adds the complexity. But again it is not well explained. I think examples based on custom styles would help a lot. Currently, the reference manual provides snippets based on the default style. Because of the complexity, this is rather difficult to figure out where the style components should be defined, where they should be used and how to put everything together in a table. I still don't know a clear distinction between `UseTblrTemplate` or `\ExpTblrTemplate`, while one can figure out the difference between `\DefTblrTemplate`, `\SetTblrTemplate`. I think in general the manual is missing step by step examples, at least for this particular section!
